velox/exec/Unnest.cpp

```diff
@@ -220,13 +220,35 @@ RowVectorPtr Unnest::generateOutput(
       // Construct unnest column using Array elements wrapped using above
       // created dictionary.
       auto unnestBaseArray = currentDecoded.base()->as<ArrayVector>();
-      outputs[outputsIndex++] = identityMapping
-          ? unnestBaseArray->elements()
-          : wrapChild(
-                numElements,
-                elementIndices,
-                unnestBaseArray->elements(),
-                nulls);
+      auto elements = unnestBaseArray->elements();
```

Comment:
I noticed that this method is very long. Perhaps, refactor to extract some helper methods.

Reply:
Created https://github.com/facebookincubator/velox/pull/8077 for the overall refactoring. I will move this logic into it.